### PR TITLE
Fix crash in README example

### PR DESCRIPTION
**Pull request by BroderickCarlin**
Thank you for helping out with embedded-graphics development! Please:

- [ ] Check that you've added passing tests and documentation
- [x] Add a `CHANGELOG.md` entry in the **Unreleased** section under the appropriate heading (**Added**, **Fixed**, etc) if your changes affect the **public API**
- [x] Run `rustfmt` on the project
- [x] Run `just build` (Linux/macOS only) and make sure it passes. If you use Windows, check that CI passes once you've opened the PR.

## PR description

Addresses https://github.com/embedded-graphics/embedded-graphics/issues/729

### CODE REVIEW

**Comment by rfuest** [comment]
IMO it would be better to replace `MockDisplay` with the new `Framebuffer`, that was added in e-g 0.8. `MockDispay` was only used in the example because no other `DrawTarget` was included in e-g at the time the example was written.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,10 @@ Embedded Graphics is a `no_std` library for adding graphics features to display
 
 ## [Unreleased] - ReleaseDate
 
+### Fixed 
+
+- [#734](https://github.com/embedded-graphics/embedded-graphics/pull/734) Fix panic if the README example is ran
+
```
Comment (rfuest) [inline CHANGELOG.md:12]: I don't think this change needs a changelog entry. Our policy had been to only include important API changes in the changelog to make it easier for users to find the relevant changes.

**Comment by BroderickCarlin** [comment]
Removed the changelog entry and updated to use the `FrameBuffer` type

File: src/lib.rs [hunk 2]
```diff
@@ -159,9 +160,15 @@
 //! };
 //!
 //! fn main() -> Result<(), std::convert::Infallible> {
-//!     // Create a new mock display
-//!     let mut display: MockDisplay<BinaryColor> = MockDisplay::new();
-//! #   display.set_allow_overdraw(true);
+//!     // Create a new mock framebuffer
```
Comment (rfuest) [inline src/lib.rs:163]:
```suggestion
//!     // Create a new framebuffer
```

**Review by rfuest — CHANGES_REQUESTED** [review]

**Comment by rfuest** [comment]
> My initial thought is that this made the example more verbose and I'm not sure it is a great README level example given the type complexity of the `FrameBuffer`.

Using `MockDisplay` also doesn't make a lot of sense IMO, because it's only intended for writing tests and won't be used by most users. They are more likely to use `Framebuffer` in real applications, but I agree that the complex type isn't great for an example.

In the past we had used the unmodified simulator example (https://github.com/embedded-graphics/examples/blob/main/eg-next/examples/hello-world.rs), which would be the more useful, but we did run into CI issues with cyclic dependencies.

**Comment by rfuest** [comment]
The paragraph above the example code still contains a reference to `MockDisplay`.

**Comment by BroderickCarlin** [comment]
Good catch on the lingering `MockDisplay` - fixed

> They are more likely to use `Framebuffer` in real applications

I think this is a fair argument and points to e-g providing more than just graphical primitives. I'm fine leaving it as-is

**Review by rfuest — APPROVED** [review]
LGTM

**Pull request merged**